### DISCUSSION TITLE
[Feature] UI enhancements

### DISCUSSION THREAD

**Discussion by Ecostack** [apache/skywalking#11479] [Ideas]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Hey all,

while using Skywalking, I found somethings that can be improved with the UI.

1. It seems like the columns in the Log view cannot be resized. This makes glancing over the logs via the Content column difficult. The current width is just 300px, which might not be enough for some cases.
2. Keep the current UI selection/filter state as part of the URL. With that, the link can easily be shared or re-opened.
3. Save trace view selection (List/Tree/Table) as part of either URL or local storage.
4. Besides give a list of logs, also show an aggregation of logs. In that case, also show how often certain log types appear.

Please let me know of these ideas.

### Are you willing to submit a pull request to implement this on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
> 4. Besides give a list of logs, also show an aggregation of logs. In that case, also show how often certain log types appear.

Could you explain more? Logs are queried on paging, how to aggregated? Usually, we recommend streaming log analysis through [LAL](https://skywalking.apache.org/docs/main/next/en/concepts-and-designs/lal/).

**Reply by Ecostack** [reply]
I am thinking similar to what Uptrace is doing:

https://app.uptrace.dev/spans/1/?time_gte=20231101T065000&time_dur=3600&system=log%3Aall&query=group%20by%20.group_id%20%7C%20per_min%28.count%29&sort_by=per_min%28.count%29&sort_desc=true&column=per_min%28.count%29


See the count per minute aggregation:
<img width="1477" alt="Screenshot 2023-11-01 at 16 06 25" src="https://github.com/apache/skywalking/assets/8505607/b04a1066-7dbb-4658-8834-8bf9f7c58ac6">

**Reply by Superskyyy** [reply]
@Ecostack This is indeed coming as an AI algorithm add-on from the aiops engine side of skywalking, it cannot be achieved using purely UI or backend through aggregation since users otherwise have to manually decide template for possiblly hundreds of unstructured logs each with dynamic variables.

**Comment by wu-sheng** [comment]
> 1. It seems like the columns in the Log view cannot be resized. This makes glancing over the logs via the Content column difficult. The current width is just 300px, which might not be enough for some cases.

If the messages are super long, you need a very wide column to read. Could you share your use cases which is inconvenient to read raw logs in the open dialog?

**Reply by Ecostack** [reply]
In my opinion, the focus of the log view should be on the content column. Giving it the most space.

In my case, the logged content contains also a time stamp, that is blocking the view. It is maybe not the best example but looking at Uptrace, it gives 2-3 times the space for the logged content.

My Skywalking log view:

<img width="1308" alt="Screenshot 2023-11-01 at 15 52 50" src="https://github.com/apache/skywalking/assets/8505607/996584c7-4c63-498b-aef1-a1736befbbac">


Uptrace demo:
<img width="1493" alt="Screenshot 2023-11-01 at 15 50 42" src="https://github.com/apache/skywalking/assets/8505607/dfd2125c-7f37-4bc5-a5cb-c9cb037df34d">

**Reply by wu-sheng** [reply]
I just put my conclusion as below, please take a look.

**Comment by wu-sheng** [comment]
@Fine0830 I would like to recommend enhancing log view page like this

1. Timestamp + Raw log(content) as column one, 70% of the initial table width. No separate timestamp column. The first column should be `timestamp` - `content`
2. Service
3. `Level` tag(`key=level`)'s value. This could be missed from tags, if so, keep the column as empty.
4. `Trace` link. If trace ID exists, generate a link to trace. Don't need to put trace ID in the text.
5. Still keep details pop up style.
6. No column resize is required when having <1> - <5>

**Reply by Fine0830** [reply]
OK

**Reply by wu-sheng** [reply]
Implementations on https://github.com/apache/skywalking-booster-ui/pull/333
